### DISCUSSION TITLE
Searching for customer without completed orders

### DISCUSSION THREAD

**Discussion by sjelfull** [craftcms/commerce#2385] [Ideas]
### Description

When creating a new order manually in Commerce, the customer search excludes customers that haven't completed any orders.

This means we won't be able to search for someone that has a customer and/or user but no orders.

**Comment by lukeholder** [comment]
Right now we don't consider a customer a customer without a completed order.

You can still type in an email address and create the customer record when the order is marked as complete.

We want to fix this in the future. We need to expand the definition of a customer (This includes someone that has a subscription but no orders).

**Comment by sjelfull** [comment]
Ah right. From a UX standpoint I was a bit unsure about what would happen when I "create" a customer, knowing that there is already a user/customer for that email in the system already.

Maybe the UX could be improved a bit to include at least customers with inactive carts on demand?

Use case I have in mind is when customers is trying to place an order, but face some issue like payment or hard to identify problems.

They might have filled out all their details including address etc.

That being said, after writing this out, that particular use case might be better served by seeking out their (in)active cart and turning it into an order.

**Comment by elivz** [comment]
Just want to add my vote for changing how this works. In our case we are syncing customer records from a 3rd-party ERP system. The customer service reps expect to be able to look up existing customers and have their addresses & other info pre-filled, regardless of their order history. It appears that even when the user and customer records exist, there is no way to access them when creating a new order, which leads to considerable confusion.
